### PR TITLE
feat(toolkit): add new Modals

### PR DESCRIPTION
**Pull request by knajjars**
https://github.com/user-attachments/assets/a36cee7f-bfd4-40a3-b368-2c65c98cad72

**AI Description**

This PR updates the useContext import to useContextStore in multiple files. It also includes changes to the Modal component, removing the kind prop and updating the styles for the DialogPanel and Modal components.

- Updates the `useContext` import to `useContextStore` in the following files:
 - Chat.tsx
 - CreateAgent.tsx
 - CitationDocumentSnippet.tsx
 - Header.tsx
 - ConversationCard.tsx
 - DeploymentsDropdown.tsx
 - EditEnvVariablesButton.tsx
- Removes the kind prop from the Modal component and updates the styles for the DialogPanel and Modal components in Modal.tsx.
- Updates the z-index values for the backdrop and modal classes in tailwind.config.js.

### CODE REVIEW

File: src/interfaces/assistants_web/src/app/layout.tsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ export const metadata: Metadata = {
 const Layout: React.FC<React.PropsWithChildren> = ({ children }) => {
   return (
     <html lang="en" className="dark">
-      <body>
+      <body className="text-volcanic-100 dark:text-marble-950">
```
Comment (tomtobac) [inline src/interfaces/assistants_web/src/app/layout.tsx:16]: we will need to update so many components now that don't require this classes anymore

File: src/interfaces/assistants_web/src/components/Shared/Modal/Modal.tsx [hunk 2]
```diff
@@ -1,86 +1,15 @@
 'use client';
 
-import { Dialog, Transition } from '@headlessui/react';
-import { cva } from 'class-variance-authority';
```
Comment (tomtobac) [inline src/interfaces/assistants_web/src/components/Shared/Modal/Modal.tsx:4]: If we remove CVA, can we remove the dependency too? 🙏

File: src/interfaces/assistants_web/src/components/Shared/Modal/Modal.tsx [hunk 3]
```diff
@@ -128,38 +52,36 @@ export const Modal: React.FC<ModalProps> = ({
...
-                      {!hideCloseButton && (
-                        <CloseButton className={cn(closeButtonStyle({ kind }))} onClose={onClose} />
-                      )}
                       {title && (
-                        <Dialog.Title as="div" className="flex-0">
-                          {typeof title === 'string' ? (
-                            <Text
-                              as="h5"
-                              className={cn(titleStyle({ kind }))}
-                              styleAs={isCitation ? 'label' : 'h5'}
```
Comment (tomtobac) [inline src/interfaces/assistants_web/src/components/Shared/Modal/Modal.tsx:145]: we don't have modal citations in mobile?

Comment (knajjars) [reply src/interfaces/assistants_web/src/components/Shared/Modal/Modal.tsx:145]: not shown in designs, but it's easy to change this with specs

File: src/interfaces/assistants_web/src/app/layout.tsx (hunk 1 above)
Comment (knajjars) [reply src/interfaces/assistants_web/src/app/layout.tsx:16]: done

**Pull request merged**